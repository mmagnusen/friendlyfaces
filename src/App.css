.Gallery {
  background: grey;
  padding: 2rem;
  display: flex;
  height: 100vh;
  justify-content: space-between; 
}

img {
  height: 550px;
  width: auto;
  border: 10px solid #FFF;
}