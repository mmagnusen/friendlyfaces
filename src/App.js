import React, { Component } from 'react';
import { Kimson, ShalomOne, ShalomTwo } from './assets'
import './App.css';

class App extends Component {
  state = {
    profiles: [{
      creator: 'Kimson Doan',
      image: Kimson,
      url: 'https://unsplash.com/@doanstopexploring'
    }, 
    {
      creator: 'Shalom Mwenesi',
      image: ShalomOne,
      url: 'https://unsplash.com/@5halom_'
    },
    {
      creator: 'Shalom Mwenesi',
      image: ShalomTwo,
      url: 'https://unsplash.com/@5halom_'
    },]
  };

  render() {

    const { profiles } = this.state;

    return (
      <div className="Gallery">
        {profiles && profiles.map((profile) => <div key={profile.image}><img src={profile.image} alt={`profile from ${profile.creator}`}/></div>)}
      </div>
    )};
}

export default App;
