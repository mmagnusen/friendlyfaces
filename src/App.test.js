import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('The Main App', () => {
  test('It renders with gallery', () => {
    const component = renderer.create(
      <App />
    )

    expect(component.toJSON()).toMatchSnapshot();
  })
})
