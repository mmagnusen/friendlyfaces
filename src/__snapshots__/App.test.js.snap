// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Main App It renders with gallery 1`] = `
<div
  className="Gallery"
>
  <div>
    <img
      alt="profile from Kimson Doan"
      src="kimson.jpg"
    />
  </div>
  <div>
    <img
      alt="profile from Shalom Mwenesi"
      src="shalom-one.jpg"
    />
  </div>
  <div>
    <img
      alt="profile from Shalom Mwenesi"
      src="shalom-two.jpg"
    />
  </div>
</div>
`;
